From ae4aea11c95b47a62f22c8f8b6cb8165be000c76 Mon Sep 17 00:00:00 2001
From: Lukasz Smurawski <lukasz.smurawski@grinn-global.com>
Date: Wed, 6 May 2026 16:32:56 +0200
Subject: [PATCH] Add grinn-astra-1680-sbc configuartion

Configure platform-related GPIOs for grinn-astra-1680-sbc.

Changes:
- Remove I2C expander 1 (N/A)
- Remove ADC-related configuration
- Retain only used GPIOs:
    - I2C expander 0 (0x43): pins 0-4
    - SM_GPIO 7

---
 .../dolphin/dolphin-rdk/platform_config.h     | 32 -------------------
 .../dolphin-rdk/platform_customization.c      | 28 ++--------------
 2 files changed, 3 insertions(+), 57 deletions(-)

diff --git a/bootloader/sm_cm3/syna/customization/dolphin/dolphin-rdk/platform_config.h b/bootloader/sm_cm3/syna/customization/dolphin/dolphin-rdk/platform_config.h
index 41af397..b3f262d 100644
--- a/bootloader/sm_cm3/syna/customization/dolphin/dolphin-rdk/platform_config.h
+++ b/bootloader/sm_cm3/syna/customization/dolphin/dolphin-rdk/platform_config.h
@@ -37,13 +37,9 @@
 // TW3, GPIO0_3, VDDM_LPQ_OFF, active high.
 // TW3, GPIO0_4, STAND-BY_EN, active high.
 
-// U13, FLX6408, ADDR = 1. I2C 7-bit address: 0x44
-// TW3, GPIO1_2, WL_REG_ON#, active low.
-
 /* Power pins definitions*/
 #define I2C_MASTER_ID 	1
 #define I2C_SLAVE_ADDR 	0x43
-#define I2C_SLAVE1_ADDR 0x44
 
 /* I2C expander GPIO0 */
 #define GPIO_VCORE 	0
@@ -66,40 +62,12 @@
 #define PERI_ON 	0
 #define PERI_OFF 	1
 
-/* I2C expander GPIO1 */
-#define GPIO_WIFI	2
-#define WIFI_ON 	1
-#define WIFI_OFF 	0
-
 #define SM_GPIO_PORT_DDR_CTRL GPIO_VDDM
 
-/* WiFi GPIO definitions*/
-#define SM_GPIO_PORT_WIFIBT 7
 #define SM_GPIO_PORT_GEPHY 7
 
 #define WIFIBT_GAP_MIN	10
 #define WIFIBT_GAP_MAX	125
 #define ETH_GAP_MAX	512
 
-/* ADC button press time threshold*/
-#define ADC_BTN_SW1_SHORT_PRESS_ACTION_TIME	500
-#define ADC_BTN_SW1_LONG_PRESS_ACTION_TIME	5*1000
-#define ADC_BTN_SW2_SHORT_PRESS_ACTION_TIME	100
-#define ADC_BTN_SW2_LONG_PRESS_ACTION_TIME	5*1000
-#define ADC_BTN_SW3_SHORT_PRESS_ACTION_TIME	100
-#define ADC_BTN_SW3_LONG_PRESS_ACTION_TIME	5*1000
-#define ADC_BTN_SW4_SHORT_PRESS_ACTION_TIME	100
-#define ADC_BTN_SW4_LONG_PRESS_ACTION_TIME	5*1000
-#define LONG_PRESS_TIMEOUT			10*1000
-
-/* ADC button IR Key */
-#define ADC_BTN_SW1_SHORT_PRESS_KEY	MV_IR_KEY_BEGIN_PAIRING
-#define ADC_BTN_SW1_LONG_PRESS_KEY 	MV_IR_KEY_NULL
-#define ADC_BTN_SW2_SHORT_PRESS_KEY	MV_IR_KEY_POWER
-#define ADC_BTN_SW2_LONG_PRESS_KEY 	MV_IR_KEY_RECOVERY
-#define ADC_BTN_SW3_SHORT_PRESS_KEY	MV_IR_KEY_VOL_MINUS
-#define ADC_BTN_SW3_LONG_PRESS_KEY 	MV_IR_KEY_NULL
-#define ADC_BTN_SW4_SHORT_PRESS_KEY	MV_IR_KEY_VOL_PLUS
-#define ADC_BTN_SW4_LONG_PRESS_KEY 	MV_IR_KEY_NULL
-
 #endif /* __PLATFORM_CONFIG_H__ */
diff --git a/bootloader/sm_cm3/syna/customization/dolphin/dolphin-rdk/platform_customization.c b/bootloader/sm_cm3/syna/customization/dolphin/dolphin-rdk/platform_customization.c
index 84bc1b2..0a69a42 100644
--- a/bootloader/sm_cm3/syna/customization/dolphin/dolphin-rdk/platform_customization.c
+++ b/bootloader/sm_cm3/syna/customization/dolphin/dolphin-rdk/platform_customization.c
@@ -89,43 +89,21 @@ void board_soc_poweroff(void)
 
 void board_wifi_poweron(void)
 {
-	if(sm_fxl16408_gpios_write(I2C_MASTER_ID, I2C_SLAVE1_ADDR,
-		(1 << GPIO_WIFI), (WIFI_ON << GPIO_WIFI)) != 0)
-		PRT_ERROR("%s: failed\n", __func__);
+	return;
 }
 
 void board_wifi_poweroff(void)
 {
-	if(sm_fxl16408_gpios_write(I2C_MASTER_ID, I2C_SLAVE1_ADDR,
-		(1 << GPIO_WIFI), (WIFI_OFF << GPIO_WIFI)) != 0)
-		PRT_ERROR("%s: failed\n", __func__);
+	return;
 }
 
-DECLARE_GPIO_ISR(SM_GPIO_PORT_WIFIBT, EDGE_SENSITIVE, FALLING_EDGE, sm_gpio_isr);
-
-#ifndef WIFIBT_GAP_MIN
-#define WIFIBT_GAP_MIN	10
-#define WIFIBT_GAP_MAX	125
-#define ETH_GAP_MAX	512
-#endif
+DECLARE_GPIO_ISR(SM_GPIO_PORT_GEPHY, EDGE_SENSITIVE, FALLING_EDGE, sm_gpio_isr);
 
 #ifndef BTKEY_NETFLIX_GAP_MIN
 #define BTKEY_NETFLIX_GAP_MIN 395
 #define BTKEY_NETFLIX_GAP_MAX 415
 #endif
 
-/* wifi */
-DECLARE_GPIO_WAKEUP(wifi, SM_GPIO_PORT_WIFIBT, 0, (WIFIBT_GAP_MIN *2),
-							MV_SM_WAKEUP_SOURCE_WIFI, 0, 0);
-
-/* bt (not netflix key) */
-DECLARE_GPIO_WAKEUP(bt_otherkey, SM_GPIO_PORT_WIFIBT, (WIFIBT_GAP_MIN *2), BTKEY_NETFLIX_GAP_MIN,
-							MV_SM_WAKEUP_SOURCE_BT, 1, KEY_OTHERS);
-
-/* bt (netflix key) */
-DECLARE_GPIO_WAKEUP(bt_netflixkey, SM_GPIO_PORT_WIFIBT, BTKEY_NETFLIX_GAP_MIN, BTKEY_NETFLIX_GAP_MAX,
-							MV_SM_WAKEUP_SOURCE_BT, 1, KEY_NETFLIX);
-
 /* geth */
 DECLARE_GPIO_WAKEUP(geth, SM_GPIO_PORT_GEPHY, BTKEY_NETFLIX_GAP_MAX, DEFAULT_THOLD,
 							MV_SM_WAKEUP_SOURCE_WOL, 0, 0);
-- 
2.43.0

